### PR TITLE
Use print_svg in step 2

### PR DESCRIPTION
**Pull request by tcclevenger**
Changed step 2 to output sparsity pattern using .svg file instead of GNU plot and changed documentation to reflect this change. Reasons for this change are nicer output and it is easier to look at than GNU plot.

### CODE REVIEW

**Comment by QiaoLei-88** [comment]
I suggest to move this new feature to the  ```Possible extensions``` section.
Because after this change, the reader will loose a change to learn how to generate result file for GNU plot.

File: examples/step-2/doc/results.dox [hunk 1]
```diff
@@ -1,33 +1,7 @@
...
-
-        Send comments and requests for help to <info-gnuplot@dartmouth.edu>
-        Send bugs, suggestions and mods to <bug-gnuplot@dartmouth.edu>
-
-
-Terminal type set to 'x11'
-gnuplot> set style data points
-gnuplot> plot "sparsity_pattern.1"
-@endcode
+patterns. We can visualize them by opening the .svg files in a web browser.
```
Comment (bangerth) [inline examples/step-2/doc/results.dox]: Can you mark this up as `<code>.svg</code>`?

File: examples/step-2/doc/results.dox [hunk 2]
```diff
@@ -38,11 +12,11 @@ equation):
 <TABLE WIDTH="60%" ALIGN="center">
   <tr>
     <td ALIGN="center">
-      <img src="http://www.dealii.org/images/steps/developer/step-2.sparsity-1.png" alt="">
+      @image html sparsity_pattern1.svg
```
Comment (bangerth) [inline examples/step-2/doc/results.dox]: These will eventually have to move to the server. Please use `http://www.dealii.org/images/steps/developer/step-2.sparsity-1.svg` and similarly for the second one below.

File: examples/step-2/doc/results.dox [hunk 3]
```diff
@@ -76,3 +50,31 @@ i.e. the matrix clusters more around the diagonal.
 Another idea of experiments would be to try other renumbering
 strategies than Cuthill-McKee from the DoFRenumbering namespace and see how
 they affect the sparsity pattern.
+
+You can also visualize the output using <a
+href="http://www.gnuplot.info/">GNUPLOT</a> (one of the simpler visualization
+programs; maybe not the easiest to use since it is command line driven, but
+also universally available on all Linux and other Unix-like systems) by changing from "print_svg()" to "print_gnuplot" in distribute_dofs() and renumber_dofs():
```
Comment (bangerth) [inline examples/step-2/doc/results.dox]: mark up all of the calls using `<code>...</code>` (and no quotes). Be consistent about using parentheses after the function name.

File: examples/step-2/step-2.cc [hunk 4]
```diff
@@ -85,13 +85,13 @@ void make_grid (Triangulation<2> &triangulation)
                outer_radius = 1.0;
   GridGenerator::hyper_shell (triangulation,
                               center, inner_radius, outer_radius,
-                              10);
+                              5 );
```
Comment (bangerth) [inline examples/step-2/step-2.cc]: Why this change?

File: examples/step-2/step-2.cc [hunk 5]
```diff
@@ -85,13 +85,13 @@ void make_grid (Triangulation<2> &triangulation)
...
 
   static const SphericalManifold<2> manifold_description(center);
   triangulation.set_all_manifold_ids(0);
   triangulation.set_manifold (0, manifold_description);
 
-  for (unsigned int step=0; step<5; ++step)
+  for (unsigned int step=0; step<3; ++step)
```
Comment (bangerth) [inline examples/step-2/step-2.cc]: And why this one?

File: examples/step-2/step-2.cc [hunk 6]
```diff
@@ -216,9 +216,9 @@ void distribute_dofs (DoFHandler<2> &dof_handler)
...
 
   // With this, we can now write the results to a file:
-  std::ofstream out ("sparsity_pattern.1");
-  sparsity_pattern.print_gnuplot (out);
-  // The result is in GNUPLOT format, where in each line of the output file,
+  std::ofstream out ("sparsity_pattern1.svg");
+  sparsity_pattern.print_svg (out);
+  // The result is in .svg format, where in each line of the output file,
   // the coordinates of one nonzero entry are listed. The output will be shown
   // below.
```
Comment (bangerth) [inline examples/step-2/step-2.cc]: This text doesn't match any more. Please adjust the whole paragraph.

**Comment by bangerth** [comment]
I actually like this. `gnuplot` is a bit old-fashioned, so using a more modern approach is probably the better idea. The old text remains in place, after all.

Can you make the suggested modifications? Thanks a lot in advance!

**Comment by tjhei** [comment]
> I actually like this.

Glad you agree. Installing/using gnuplot was one of the early hurdles many of my students encountered.

File: examples/step-2/step-2.cc (hunk 4 above)
Comment (tjhei) [inline examples/step-2/step-2.cc]: A coarser mesh makes the matrix smaller, which makes the pixels in the pattern much easier to see.

File: examples/step-2/doc/results.dox (hunk 2 above)
Comment (bangerth) [inline examples/step-2/doc/results.dox]: OK, I've uploaded both sparsity pattern files to the mentioned location.

File: examples/step-2/step-2.cc (hunk 4 above)
Comment (bangerth) [inline examples/step-2/step-2.cc]: Fair enough. Can you mention in the comment above the `make_grid()` function that it is the same mesh as in step-1, but with fewer refinement steps?

Comment (tcclevenger) [inline examples/step-2/step-2.cc]: Sorry for the late reply, but yes I can see to all these changes. I'll let
you know when I'm finished.

Conrad

On Thu, Sep 17, 2015 at 8:13 AM, Wolfgang Bangerth <notifications@github.com

**Comment by tcclevenger** [comment]
All changes have been made. Let me know if there are any issues.
